validator/sawtooth_validator/journal/publisher.py

```diff
@@ -490,8 +494,9 @@ def on_check_publish_block(self, force=False):
                         self._pending_batches:
                     self._build_candidate_block(self._chain_head)
 
-                if self._candidate_block and \
-                        (force or self._pending_batches) and \
+                if self._candidate_block and (
+                            force or
+                            self._candidate_block.has_pending_batches()) and \
```

Comment:
Given that if `_pending_batches` is empty, it evaluates to false, this doesn't necessarily fix the cause as to why `last_batch` was being called in the first place.  The error message from last_batch might be slightly more informative, though.